### DISCUSSION TITLE
fish vi cursor?

### DISCUSSION THREAD

**Discussion by beaumccartney** [wezterm/wezterm#4670] [Q&A]
### What Operating System(s) are you running on?

macOS

### WezTerm version

wezterm 20231203-124028-e3cd2e93

### Ask your question!

How can I get wezterm to respect `fish_vi_cursor`? I tried `set fish_vi_force_cursor` and setting the vte num but still not responsive. Just sticks with whatever cursor I put in config or default if nothing.

**Comment by wez** [comment]
wezterm doesn't care if fish is running; it just follows the escape sequences that are output to it. I think this is most likely a question for the fish folks to troubleshoot what that option does and why it doesn't seem to take effect in wezterm.

**Comment by januz** [comment]
@BeauSLM I just switched from kitty (where this works) to WezTerm and am experiencing the same issue. Did you get more information from the fish developers?

**Reply by beaumccartney** [reply]
Unfortunately I haven't had time to look into the issue. I probably won't at least until the new year.

**Reply by januz** [reply]
@BeauSLM I found an old [issue](https://github.com/wez/wezterm/issues/2781#issuecomment-1324143452) that has the solution:

Add `set fish_vi_force_cursor 1` to your `.config/fish/config.fish`

**Reply by andradei** [reply]
@januz Thanks! Good thing people found that. It's buried on the [fish docs](https://fishshell.com/docs/current/interactive.html)
